File: 3803-count-residue-prefixes/3803-count-residue-prefixes.go
func residuePrefixes(s string) int {
    ans, dis := 0, 0
    freq := make([]int, 26)

    for i, ch := range s {
        id := int(ch - 'a')
        if freq[id] == 0 {
            dis++
        }
        
        freq[id]++
        if (dis == (i + 1) % 3) {
            ans++
        }
    }
    return ans
}

File: 3803-count-residue-prefixes/3803-count-residue-prefixes.c
int residuePrefixes(char* s) {
    int dis = 0, i = 0, ans = 0, n = strlen(s);
    int freq[26] = {};

    for (; *s; s++) {
        int ch = *s - 'a';
        if (freq[ch] == 0)
            dis++;
        freq[ch]++;
        ans += (dis == (i + 1) % 3);
        i++;
    }
    return ans;
}